### ISSUE TITLE
Cannot read property 'node' of null

### ISSUE THREAD

**Issue by niieani** [decaffeinate/decaffeinate#544]
decaffeinate is crashing on my CoffeeScript input:

```coffee
rethinkdb.monday = new (class extends RDBConstant then tt: protoTermType.MONDAY, st: 'monday')()
```

[(repl)](http://decaffeinate-project.org/repl/#?evaluate=false&stage=full&code=rethinkdb.monday%20%3D%20new%20(class%20extends%20RDBConstant%20then%20tt%3A%20protoTermType.MONDAY%2C%20st%3A%20'monday')())

I get this error:

```
Cannot read property 'node' of null
> 1 | rethinkdb.monday = new (class extends RDBConstant then tt: protoTermType.MONDAY, st: 'monday')()
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by alangpierce** [comment]
Wow, I didn't know you could do that...

But it looks like that kind of code also works in JS, so hopefully should be reasonable to fix.

**Comment by niieani** [comment]
Actually its not my code, but I believe class fields would be ok there. 
The stricter option doesn't seem so bad, and we should keep in mind that without it someone could run into some problematic cases.

**Comment by alangpierce** [comment]
Ok, sounds good, I think I'll attempt the second option when I get a chance, and if people care, then at some point there can be an option to use class fields.

**Comment by niieani** [comment]
Cool. It would make sense to have something like an optional _loose-mode_, similarly to babel. That could possibly go into that.

**Comment by niieani** [comment]
Then there's the [fields and static properties proposal](https://github.com/tc39/proposal-class-public-fields), which is already TC Stage 2. 

Then we could simply:

```
rethinkdb.monday = new (class extends RDBConstant {
  static tt = protoTermType.MONDAY;
  static st = 'monday';
})()
```

**Comment by mbroadst** [comment]
FYI coffeescript compiles this to the following:
```
  rethinkdb.monday = new ((function(superClass) {
    extend(_Class, superClass);

    function _Class() {
      return _Class.__super__.constructor.apply(this, arguments);
    }

    _Class.prototype.tt = protoTermType.MONDAY;

    _Class.prototype.st = 'monday';

    return _Class;

  })(RDBConstant))();
```

**Comment by niieani** [comment]
Great stuff, thank you @alangpierce.

**Comment by alangpierce** [comment]
👍 . Also, oops, apparently github auto-closed the issue because I wrote "almost fixes #544". The main piece is there, but this specific example still breaks, so I'll leave this specific task open. We just need to be a little smarter about commas within class bodies for the original code example to completely work.
